tests/model_registry/rbac/test_mr_rbac.py

```diff
@@ -0,0 +1,248 @@
+import pytest
+from typing import Self, Callable, ContextManager
+import os
+from simple_logger.logger import get_logger
+from contextlib import nullcontext
+
+from model_registry import ModelRegistry as ModelRegistryClient
+from tests.model_registry.constants import MR_INSTANCE_NAME, MR_NAMESPACE
+from tests.model_registry.rbac.utils import get_token, build_mr_client_args
+from tests.model_registry.utils import get_endpoint_from_mr_service, get_mr_service_by_label
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.model_registry import ModelRegistry
+from ocp_resources.role import Role
+from ocp_resources.role_binding import RoleBinding
+from utilities.constants import DscComponents, Protocols
+from mr_openapi.exceptions import ForbiddenException
+
+
+LOGGER = get_logger(name=__name__)
+NEW_GROUP_NAME = "test-model-registry-group"
+
+
+def assert_positive_mr_registry(
+    model_registry_instance: ModelRegistry,
+    model_registry_namespace: str,
+    admin_client: DynamicClient,
+    user_name: str,
+    password: str,
+) -> None:
+    """
+    Assert that a user has access to the Model Registry.
+    """
+    user_token = get_token(user_name=user_name, password=password, admin_client=admin_client)
+    namespace_instance = admin_client.resources.get(api_version="v1", kind="Namespace").get(
+        name=model_registry_namespace
+    )
+    svc = get_mr_service_by_label(client=admin_client, ns=namespace_instance, mr_instance=model_registry_instance)
+    endpoint = get_endpoint_from_mr_service(svc=svc, protocol=Protocols.REST)
+    client_args = build_mr_client_args(rest_endpoint=endpoint, token=user_token, author="rbac-test-user-granted")
+    mr_client = ModelRegistryClient(**client_args)
+    assert mr_client is not None, "Client initialization failed after granting permissions"
+    LOGGER.info("Client instantiated successfully after granting permissions.")
+
+
+@pytest.mark.parametrize(
+    "updated_dsc_component_state_scope_class",
+    [
+        pytest.param({
+            "component_patch": {
+                DscComponents.MODELREGISTRY: {
+                    "managementState": DscComponents.ManagementState.MANAGED,
+                    "registriesNamespace": MR_NAMESPACE,
+                },
+            }
+        })
+    ],
+    indirect=True,
+    scope="class",
+)
+@pytest.mark.usefixtures("updated_dsc_component_state_scope_class")
+class TestUserPermission:
+    """
+    Test suite for verifying user and group permissions for the Model Registry.
+    """
+
+    @pytest.mark.smoke
+    @pytest.mark.parametrize(
+        "user_name, password, context_manager",
+        [
+            ("htpasswd-cluster-admin-user", os.environ.get("ADMIN_PASSWORD"), nullcontext()),
+            ("ldap-user1", os.environ.get("NON_ADMIN_PASSWORD"), pytest.raises(ForbiddenException)),
+        ],
+    )
```

Comment:
We don't need to parameterize the tests. We only have 1 test here.

Reply:
file changed now but the concept is the same, same test with admin and non admin user to verify that the first as the access and the other one no